File: Data Structures & Algorithms/first-missing-positive/submission-0.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)
        for i in range(n):
            while nums[i]-1 >= 0 and nums[i]-1 < n:
                swap1 = i
                swap2 = nums[i] - 1
                if swap1 != swap2 and nums[swap1] != nums[swap2]:
                    nums[swap1], nums[swap2] = nums[swap2], nums[swap1]
                else:
                    break
        for i in range(n):
            if i+1 != nums[i]:
                return i + 1
        return nums[-1] + 1

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        const int n = static_cast<int>(nums.size());
        int duplicates = 0;
        for (int i = 0; i < n; i++) {
            while (i+1 < n && nums[i+1] == nums[i]) {
                nums[i] = -101;
                duplicates ++;
                i++;
            }
        }
        int u = 0;
        for (int i = 0; i < n; i++) {
            while (u < n && nums[u] == -101) {
                u++;
            }
            if (u < n) {
                nums[i] = nums[u];
            }
            u++;
        }
        return n-duplicates;
    }
};